transmogrifier/sources/transformer.py

```diff
@@ -5,28 +5,49 @@
 import json
 import logging
 import os
+import re
 import uuid
 from abc import ABC, abstractmethod
+from dataclasses import dataclass
 from importlib import import_module
 from typing import TYPE_CHECKING, final
 
+import pyarrow as pa
+import pyarrow.dataset as ds
 import smart_open  # type: ignore[import-untyped]
 from attrs import asdict
+from bs4 import Tag  # type: ignore[import-untyped]
 
 import transmogrifier.models as timdex
-from transmogrifier.config import SOURCES
+from transmogrifier.config import SOURCES, get_etl_version
 from transmogrifier.exceptions import DeletedRecordEvent, SkippedRecordEvent
 from transmogrifier.helpers import generate_citation, validate_date
 
 if TYPE_CHECKING:
     from collections.abc import Callable, Iterator
 
-    from bs4 import Tag  # type: ignore[import-untyped]
+    from transmogrifier.models import TimdexRecord
 
 logger = logging.getLogger(__name__)
 
 type JSON = dict[str, "JSON"] | list["JSON"] | str | int | float | bool | None
 
+PARQUET_DATASET_BATCH_SIZE = 1_000
+
+
+@dataclass
+class ETLRecord:
```

Comment:
In retrospect, probably could have / should have marked this as "WIP" as well.  My hunch is that this approach could work well, but for the sake of this PR it's still a bit of a stub.